feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/input/ui/audiorecorder/PermissionDialog.kt

```diff
@@ -0,0 +1,42 @@
+package com.hedvig.android.odyssey.input.ui.audiorecorder
+
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.ui.res.stringResource
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun PermissionDialog(
+  recordAudioPermissionState: PermissionState,
+  openDialog: MutableState<Boolean>,
+  permissionTitle: String,
+  permissionMessage: String,
+) {
+  AlertDialog(
+    onDismissRequest = { openDialog.value = false },
+    title = { Text(permissionTitle) },
+    text = { Text(permissionMessage) },
+    dismissButton = {
+      TextButton(
+        onClick = { openDialog.value = false },
+      ) {
+        Text(stringResource(android.R.string.cancel))
+      }
+    },
+    confirmButton = {
+      TextButton(
+        onClick = {
+          openDialog.value = false
+          recordAudioPermissionState.launchPermissionRequest()
+        },
+      ) {
+        Text(stringResource(android.R.string.ok))
+      }
+    },
+  )
+}
```

Comment:
I'd propose to change this to
```suggestion
fun PermissionDialog(
  recordAudioPermissionState: PermissionState,
  permissionTitle: String,
  permissionMessage: String,
  dismiss: () -> Unit,
) {
  AlertDialog(
    onDismissRequest = { dismiss() },
    title = { Text(permissionTitle) },
    text = { Text(permissionMessage) },
    dismissButton = {
      TextButton(
        onClick = { dismiss() },
      ) {
        Text(stringResource(android.R.string.cancel))
      }
    },
    confirmButton = {
      TextButton(
        onClick = {
          dismiss()
          recordAudioPermissionState.launchPermissionRequest()
        },
      ) {
        Text(stringResource(android.R.string.ok))
      }
    },
  )
}
```

This dialog doesn't need to know about the status of this variable, in fact it should only be shown when it's true, it never does anything when the value is false, cause on the call-site it's simply not called therefore not shown on screen.
It just needs to know that it can dismiss itself, and on the call-site you can call this like:
```kotlin
  if (openPermissionDialog.value) {
    PermissionDialog(
      recordAudioPermissionState,
      "Permission required",
      "We need permission to record audio",
      dismiss = { openPermissionDialog = false }
    )
  }
```

Reply:
Excellent, fixing 👍